src/main/scala/cromwell/engine/backend/io/package.scala

```diff
@@ -0,0 +1,58 @@
+package cromwell.engine.backend
+
+import java.nio.file._
+
+import cromwell.engine.backend.io.filesystem.gcs.{ContentTypeOption, GcsFileSystem, GcsFileSystemProvider, NioGcsPath}
+
+import scala.util.{Success, Try}
+
+package object io {
+  object PathString {
+    val defaultGCSFileSystem = GcsFileSystem.defaultGcsFileSystem
+    val defaultFileSystem = FileSystems.getDefault
+    private val defaultFileSystemsList = List(defaultGCSFileSystem, defaultFileSystem)
+
+    implicit class UriString(val str: String) extends AnyVal {
+
+      def isGcsUrl: Boolean = str.startsWith("gs://")
+      def isUriWithProtocol: Boolean = "^[a-z]+://".r.findFirstIn(str).nonEmpty
+
+      def toPath(implicit fss: List[FileSystem] = defaultFileSystemsList): Path = {
```

Comment:
Yeah, I don't really know what else to call it as it effectively takes a String and builds a Path :/
The filesystems don't have to be `implicit`s, but I think this is a pretty good use case for i, because when using this functionality, you're likely to have a list of available file systems created at a high level, and then to want to use them whenever you want to build a `Path`. If they're implicit you don't have to pass them in every single call to `toPath`, and if you do want to pass them (or others) you still can.

However I know I'm definitely part of a minority here with implicits so I'm not going to argue for ages for it :p